packages/mobx-state-tree/src/types/complex-types/map.ts

```diff
@@ -141,7 +142,7 @@ export enum MapIdentifierMode {
 
 class MSTMap<IT extends IAnyType> extends ObservableMap<string, any> {
     constructor(
-        initialData?: [string, any][] | IKeyValueMap<any> | Map<string, any> | undefined,
+        initialData?: IObservableMapInitialValues<string, any> | undefined,
```

Comment:
@jamonholmgren - Would a change to TypeScript typings like this technically be a breaking change? I think it's the right move to make, but I"m wondering if we should get some backwards compatibility here.

@BrianHung - what do you think about typing `initialData` with another union that includes `IObservableMapInitialValues`, but allows people to keep using older typings they were before?

Reply:
Double check me on this but I thought it was backwards compatible?

```ts
export type IObservableMapInitialValues<K = any, V = any> =
    | IMapEntries<K, V>
    | IReadonlyMapEntries<K, V>
    | IKeyValueMap<V>
    | Map<K, V>

export type IMapEntry<K = any, V = any> = [K, V]
export type IReadonlyMapEntry<K = any, V = any> = readonly [K, V]
export type IMapEntries<K = any, V = any> = IMapEntry<K, V>[]
export type IReadonlyMapEntries<K = any, V = any> = IReadonlyMapEntry<K, V>[]
```

https://github.com/mobxjs/mobx/blob/main/packages/mobx/src/types/observablemap.ts#L45-L48

Reply:
Oh nice!

That's exactly the info I needed on this one. Thank you!

Would you mind adding a test to `packages/mobx-state-tree/__tests__/core/map.test.ts` that checks for the typing here and demonstrates it works? You could follow the pattern Jamon did in https://github.com/mobxjs/mobx-state-tree/pull/2059, where he wrote a test that exercised the basic functionality, and if you modify the typing in the source, or modify the input in the test to mismatch the type, TS will throw an error.